### PR TITLE
Avoid parsing errors twice when saving CartoCSS

### PR DESCRIPTION
**Pull request by rubenmoya**
Related to: https://github.com/CartoDB/cartodb/issues/11805

The errors where already parsed when doing `new ParserCSS`, so calling again to `parser.parseErrors` led to an empty array.

### CODE REVIEW

**Comment by rubenmoya** [comment]
Totally true! I added a new spec file with the tests related to the `parseError` function

File: lib/assets/test/spec/builder/helpers/parser-css.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      var errors = parser.parseError([
+        'style.mss:7:2 General Kenobi',
+        'style.mss:3:2 Hello there'
+      ]);
+
+      expect(errors[0].message).toEqual('Hello there');
+      expect(errors[1].message).toEqual('General Kenobi');
+    });
+
+    it('removes duplicated errors', function () {
```
Comment (VictorVelarde) [inline lib/assets/test/spec/builder/helpers/parser-css.spec.js]: 👍 great

File: lib/assets/javascripts/builder/helpers/parser-css.js [hunk 2]
```diff
@@ -132,34 +134,23 @@ CartoParser.prototype = {
...
+    var parsedErrors = _.compact(errors).map(function (error) {
+      if (_.isObject(error)) return error;
+
+      var matchedError = error.match(ERROR_REGEX);
+      return matchedError
+        ? { line: parseInt(matchedError[1], 10), message: matchedError[3] }
+        : { line: null, message: error };
+    });
 
     // sort by line
```
Comment (VictorVelarde) [inline lib/assets/javascripts/builder/helpers/parser-css.js]: Maybe this could be a little function `_sortByLineAndRemoveDups`. Everytime I write a comment I think later on "that could be extracted"

File: lib/assets/test/spec/builder/helpers/parser-css.spec.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    describe('when the error is an object', function () {
+      it('returns the same error', function () {
+        var given = [{ line: 66, message: 'Execute order' }];
+
+        expect(parser.parseError(given)).toEqual(given);
+      });
+    });
+
+    it('returns the errors sorted by line order', function () {
+      var errors = parser.parseError([
```
[the remark this answers is not in the document]
Comment (rubenmoya) [reply lib/assets/test/spec/builder/helpers/parser-css.spec.js]: I added a new "error" which will have an empty line number to check is sorted too. About the empty list, if it's empty there is nothing to sort.

File: lib/assets/javascripts/builder/helpers/parser-css.js (hunk 2 above)
Comment (rubenmoya) [reply lib/assets/javascripts/builder/helpers/parser-css.js]: It's only used there, I don't know how useful will it be to extract it tbh

File: lib/assets/test/spec/builder/helpers/parser-css.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply lib/assets/test/spec/builder/helpers/parser-css.spec.js]: Great! I was just thinking about another test: "does the parser breaks when the error is empty"?

[the remark this answers is not in the document]
Comment (rubenmoya) [reply lib/assets/test/spec/builder/helpers/parser-css.spec.js]: Done!

File: lib/assets/test/spec/builder/helpers/parser-css.spec.js [hunk 4]
```diff
@@ -48,5 +48,17 @@ describe('helpers/parser-css', function () {
...
       expect(parsed.length).toBe(1);
     });
+
+    describe('when the error list is empty', function () {
+      it('returns an empty array', function () {
+        expect(parser.parseError([])).toEqual([]);
+      });
+    });
+
+    describe('when the error list is missisng', function () {
```
Comment (VictorVelarde) [inline lib/assets/test/spec/builder/helpers/parser-css.spec.js]: ✨  , just a typo in "missing" 👀

Comment (rubenmoya) [reply lib/assets/test/spec/builder/helpers/parser-css.spec.js]: Ooops 😅

**Comment by rubenmoya** [comment]
retest this please

**Comment by rubenmoya** [comment]
retest this please

**Comment by rubenmoya** [comment]
retest this please
